core/src/io/anuke/mindustry/maps/missions/MissionWithStartingCore.java

```diff
@@ -0,0 +1,56 @@
+package io.anuke.mindustry.maps.missions;
+
+import com.badlogic.gdx.math.GridPoint2;
+import io.anuke.mindustry.content.blocks.StorageBlocks;
+import io.anuke.mindustry.game.Team;
+import io.anuke.mindustry.maps.generation.Generation;
+import io.anuke.mindustry.world.Tile;
+
+import static io.anuke.mindustry.Vars.state;
+
+public abstract class MissionWithStartingCore extends Mission{
```

Comment:
This class needs to override `getSpawnPoints()` and return this point in the list so the area around them is generated properly.

Now that I think about it, generating cores should *only* be done using the points from `getSpawnPoints()` to streamline things. I could implement this myself later, or you could include it in this PR.

Reply:
Makes sense; I included the change in the PR